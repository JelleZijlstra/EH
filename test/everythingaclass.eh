#!/usr/bin/ehi
include '../lib/exception.eh'

const foo = 3
echo foo
# should generate an error
rescue () => (foo = 4)
echo foo
class Bar {
	echo 5
	mah = 42
}
printvar Bar.new()
