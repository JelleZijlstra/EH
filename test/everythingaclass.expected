3
Cannot set constant member foo in object of type Function: func: -> <user code>
3
5
@object <Bar> [
compare <public,non-static,non-constant>: @function <library>: 
finalize <public,non-static,non-constant>: @function <library>: 
inherit <public,non-static,non-constant>: @function <library>: 
initialize <public,non-static,non-constant>: @function <library>: 
isA <public,non-static,non-constant>: @function <library>: 
mah <public,non-static,non-constant>: @int 42
new <public,non-static,non-constant>: @function <library>: 
operator_compare <public,non-static,non-constant>: @function <library>: 
operator_equals <public,non-static,non-constant>: @function <library>: 
operator_gt <public,non-static,non-constant>: @function <library>: 
operator_gte <public,non-static,non-constant>: @function <library>: 
operator_lt <public,non-static,non-constant>: @function <library>: 
operator_lte <public,non-static,non-constant>: @function <library>: 
operator_ne <public,non-static,non-constant>: @function <library>: 
toString <public,non-static,non-constant>: @function <library>: 
]
